src/modules/UI/scenes/ChangeMiningFee/components/CustomFees/CustomFeesModal.ui.js

```diff
@@ -0,0 +1,89 @@
+// @flow
+
+import React, {Component} from 'react'
+import { View } from 'react-native'
+import StylizedModal from '../../../../components/Modal/Modal.ui'
+import { FormField } from '../../../../../../components/FormField.js'
+import * as Constants from '../../../../../../constants/indexConstants.js'
+import styles from './style'
+import OptionIcon from '../../../../components/OptionIcon/OptionIcon.ui'
+import OptionButtons from '../../../../components/OptionButtons/OptionButtons.ui.js'
+import s from '../../../../../../locales/strings.js'
+
+type Props = {
+  customFeeSettings: Array<string>,
+  visibilityBoolean: boolean,
+  onPositive: (customFees: any) => any,
+  onDone: () => any,
+  handlePress: Function
+}
+
+type State = any
+
+export default class CustomFeesModal extends Component<Props, State> {
+  constructor (props: Props) {
+    super(props)
+    this.state = {}
+  }
+
+  componentWillMount = () => {
+    this._initState()
+  }
+
+  _onFeeSettingInputChange = (feeSetting) => (input: string) => {
+    let setting = '0'
+    if (!isNaN(input) && input !== '') {
+      setting = parseInt(input).toString()
+    }
+    this.setState({ [feeSetting]: setting })
+  }
+
+  _initState = () => {
+    for (const feeSetting of this.props.customFeeSettings) {
+      this.setState({ [feeSetting]: '0' })
+    }
+  }
+
+  renderModalMiddle = () => {
+    const rows = []
+    for (const feeSetting of this.props.customFeeSettings) {
```

Comment:
Could you use `map` here?

Reply:
I changed to map.
But just in general, map is slower then for (... of ..) but since the array is tiny here it dons't really impact performance and map does look better.